hr_expense_adj/models/hr_expense.py

```diff
@@ -84,3 +85,15 @@ def _compute_account_id(self):
                 'expense']
             if account:
                 exp.account_id = account
+
+    @api.constrains('date', 'analytic_account_id')
+    def _validate_analytic_date(self):
+        for rec in self:
+            projects = rec.analytic_account_id.with_context(
+                active_test=False).mapped('project_ids')
```

Comment:
@TimLai125 Can you double check that there is no situation where sudo should be used to access projects?  Maybe it's OK as it is.

Reply:
Add sudo with  #134